networkx/algorithms/sparsifiers.py

```diff
@@ -289,5 +289,7 @@ def _add_edge_to_spanner(H, residual_graph, u, v, weight):
         The edge attribute to use as distance.
     """
     H.add_edge(u, v)
+    for attr, value in residual_graph[u][v].items():
+        H[u][v][attr] = value
     if weight:
         H[u][v][weight] = residual_graph[u][v]["weight"][0]
```

Comment:
Would this take care of all these other attributes **and** the special case of the weight attribute? 
Notice that this uses `G` instead of `residual_graph` which might make a difference, but I think
it is exactly the difference we have to "fix" in the `if weight:` statement in the original code.
```suggestion
    H.add_edge(u, v, G[u][v])
```